cmake/BundledAzureSdkCpp.cmake

```diff
@@ -16,7 +16,7 @@
 # under the License.
 
 function(use_bundled_libazure SOURCE_DIR BINARY_DIR)
-    set(PC "${Patch_EXECUTABLE}" -p1 -i "${SOURCE_DIR}/thirdparty/azure-sdk-cpp-for-cpp/azure-sdk-for-cpp-old-compiler.patch")
+    set(PC git reset --hard HEAD && "${Patch_EXECUTABLE}" -p1 -i "${SOURCE_DIR}/thirdparty/azure-sdk-cpp-for-cpp/azure-sdk-for-cpp-old-compiler.patch")
```

Comment:
It seems to be a problem now that when Minifi is built with the Azure extension and `make` command is called again with no change, because of the `git reset` the Azure extension is rebuilt. It's also a problem when `make package` is called because the preinstall target of CPack reruns the build command. For example we use it in docker image builds and the CPack phase takes considerably more time after. I tried and alternative command that seems to be working, which checks if the patch has already been applied and only applies it when needed:

```cmake
set(PATCH_FILE "${SOURCE_DIR}/thirdparty/azure-sdk-cpp-for-cpp/azure-sdk-for-cpp-old-compiler.patch")
set(PC "${Patch_EXECUTABLE}" -R -p1 -s -f --dry-run -i "${PATCH_FILE}" || "${Patch_EXECUTABLE}" -p1 -i "${PATCH_FILE}")
```

Reply:
Applied your suggestion in c2da5fe. The CI failures are unrelated to this change.